internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -2141,6 +2144,51 @@ func KexecPrepare(_ runtime.Sequence, data any) (runtime.TaskExecutionFunc, stri
 
 		defer kernel.Close() //nolint:errcheck
 
+		fd := int(kernel.Fd())
+
+		if goruntime.GOARCH == "arm64" {
+			// https://git.kernel.org/pub/scm/utils/kernel/kexec/kexec-tools.git/tree/include/kexec-pe-zboot.h
+			var peZbootheaderData [24]byte
+
+			if _, err := kernel.Read(peZbootheaderData[:]); err != nil {
```

Comment:
we should use `io.ReadFull` (https://pkg.go.dev/io#ReadFull) here, as the read is not guaranteed to fill the whole buffer